packages/elements/src/pagination/index.ts

```diff
@@ -203,109 +401,54 @@ export class Pagination extends BasicElement {
    * @returns {void}
    */
   private updateButtons (): void {
-    const page = Number.parseInt(this.page, 10);
-    const firstPage = this.disabled || page <= 1;
-    const lastPage = this.disabled || page >= this.totalPage;
+    const value = this.internalValue;
+    const firstPage = this.disabled || value <= 1;
+    const nextPage = this.disabled || value >= this.internalMax;
+    const lastPage = nextPage || this.infinitePaginate;
+
     this.previousPageButton.disabled = firstPage;
     this.firstPageButton.disabled = firstPage;
-    this.nextPageButton.disabled = lastPage;
+    this.nextPageButton.disabled = nextPage;
     this.lastPageButton.disabled = lastPage;
   }
```

Comment:
I really dislike this function. If you have time it would be good to remove this and just create some getters for the buttons states.

Reply:
Already refactored